erigon-lib/state/aggregator2.go

```diff
@@ -12,13 +12,13 @@ import (
 
 // this is supposed to register domains/iis
 
-func NewAggregator2(ctx context.Context, dirs datadir.Dirs, aggregationStep uint64, db kv.RoDB, logger log.Logger) (*Aggregator, error) {
+func NewAggregator(ctx context.Context, dirs datadir.Dirs, aggregationStep uint64, db kv.RoDB, logger log.Logger) (*Aggregator, error) {
 	salt, err := getStateIndicesSalt(dirs.Snap)
 	if err != nil {
 		return nil, err
 	}
 
-	a, err := NewAggregator(ctx, dirs, aggregationStep, db, logger)
+	a, err := newAggregatorOld(ctx, dirs, aggregationStep, db, logger)
```

Comment:
the reason I named `newAggregatorOld` as NewAggregator was because I expect this file to be moved out of erigon-lib/state...so the "general aggregator creator" provided in aggregator.go was kept to NewAggregator. 

maybe atleast `newAggreagorOld` should be public because we want to pull aggregator2.go out of this module?

Reply:
Having NewAggregator and NewAggregator2 was misleading for sure. One step at a time, having unexported method we can localize all usages, then just get rid of "deprecated construct0r"